tests/unit_tests/worker/test_task_worker.py

```diff
@@ -453,12 +453,10 @@ def take_events_from_streams(
     The type for streams will be any combination of event streams each of a
     given event type, where the event type is generic:
 
-    List[
-        Union[
-            EventStream[WorkerEvent, int],
-            EventStream[DataEvent, int],
-            EventStream[ProgressEvent, int]
-        ]
+    list[
+        EventStream[WorkerEvent, int] |
+        EventStream[DataEvent, int] |
+        EventStream[ProgressEvent, int]
```

Comment:
I assume this is correct from the Union on the left, but it may be EventStream[A|B|C, int]?

Reply:
> The type for streams will be any combination of event streams each of a given event type, where the event type is generic:

To be honest I find the above sentence very difficult to parse, maybe we should we reword it?

Reply:
I'm going to say that belongs in the docs rework not this change.